### PR TITLE
F/wep 127 richtextblock refactoring

### PR DESCRIPTION
**Pull request by faebebin**
Pure refactoring. No changes to functionality nor style. Decided to put the design fixes into new PR.

### CODE REVIEW

**Review by penk85 — DISMISSED** [review]

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        </Modal.Footer>
+      </Modal>
+    </>
+  )
+}
+
+function validateURL(url: string): string | null {
+  return url
+
+  // TODO: Implement better URL validation with for support relative links.
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/buttons.tsx:133]: Can you create a ticket for this in jira?

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      onMouseDown={e => {
+        e.preventDefault()
+        toggleFormat(editor, format)
+      }}>
+      {children}
+    </ToolbarButton>
+  )
+}
+
+export function LinkFormatButton() {
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/buttons.tsx:53]: Why is this here? It looks like the LinkFormat buttons could be also in their own file.

**Review by roosnic1 — COMMENTED** [review]
Still missing a label :)

**Review by roosnic1 — COMMENTED** [review]
Why did the `packages/editor/__tests__/specs/__snapshots__/authorEditPanel.tsx.snap` snapshot loose so many line?

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+          setLinkDialogOpen(true)
+        }}
+      />
+      <Modal show={isLinkDialogOpen} size={'sm'} onHide={() => setLinkDialogOpen(false)}>
+        <Modal.Body>
+          <Form fluid={true}>
+            <FormGroup>
+              <ControlLabel>{t('blocks.richText.link')}</ControlLabel>
+              <FormControl
+                errorMessage={url && !validatedURL ? 'Invalid Link' : undefined}
```
Comment (0815pascal) [inline packages/editor/src/client/blocks/richTextBlock/buttons.tsx:103]: Is it possible to translate `Invalid Link` ? E.g. `{t('blocks.richText.invalidLInk')}`

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+          </Form>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            disabled={isDisabled}
+            onClick={() => {
+              insertLink(editor, selection, validatedURL!, title || undefined)
+              setLinkDialogOpen(false)
+            }}>
+            {t('Apply')}
```
Comment (0815pascal) [inline packages/editor/src/client/blocks/richTextBlock/buttons.tsx:121]: I think this should be `{t('blocks.richText.apply')}`

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+          </Button>
+          <Button onClick={() => setLinkDialogOpen(false)}>{t('Cancel')}</Button>
```
Comment (0815pascal) [inline packages/editor/src/client/blocks/richTextBlock/buttons.tsx:123]: Same as comment above

**Review by 0815pascal — APPROVED** [review]
Looks good, just some minor comments

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx (hunk 1 above)
Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/buttons.tsx:133]: yep

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx (hunk 2 above)
Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/buttons.tsx:53]: agree

File: packages/editor/src/client/blocks/richTextBlock/buttons.tsx (hunk 3 above)
Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/buttons.tsx:103]: yes, will update

**Comment by faebebin** [comment]
> Why did the `packages/editor/__tests__/specs/__snapshots__/authorEditPanel.tsx.snap` snapshot loose so many line?

Seems that previously emojiMart (node_modules) source code got rendered, ie. all the possible emoji names worldwide can be found in the snap. Now no more. At least one reason for many lines diff.

**Review by 0815pascal — APPROVED** [review]

**Review by roosnic1 — APPROVED** [review]
Looks very good 🎉
